src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ImportSectionNode.cs

```diff
@@ -135,7 +135,7 @@ public override void EncodeData(ref ObjectDataBuilder dataBuilder, NodeFactory f
                 dataBuilder.EmitUInt(0);
             }
 
-            if (_emitGCRefMap)
+            if (_emitGCRefMap && !_gcRefMap.IsEmpty)
```

Comment:
```suggestion
            if (!_gcRefMap?.IsEmpty)
```
Nit: We do not really need `_emitGCRefMap`. The few places where it is used can be replaced with _gcRefMap null check.